index.ts

```diff
@@ -0,0 +1,136 @@
+class TODO {
+    nume:string;
+    data:Date;
+    stare:string;
+    descriere:string;
+    constructor(nume:string,data:Date,stare:string,descriere:string) {
+        this.nume=nume;
+        this.data=data;
+        this.stare=stare;
+        this.descriere=descriere;
+    }
+
+    finalizare(stare:string='finalizat') {
+        if(this.stare!='finalizat')
+            this.stare=stare
+        else
+            console.log('task-ul selectat este deja finalizat');
+    }
+
+    modificare(modificare:any,nume_element_modificat:string) {
+
+        switch(nume_element_modificat) {
+            case 'nume':
+                this.nume=modificare;
+                break;
+            case 'data':
+                this.data=modificare;
+                break;
+            case 'descriere':
+                this.descriere=modificare;
+                break;
+            default:
+                console.log('nu exista elemenul pe care doriti sa il modificati');
+                break;
+        }
+    }
+}
+
+class TODOS {
+tasks:Array<TODO>;
+constructor() {
+        this.tasks=[];
+    }
+adaugareTask(task:TODO) {
+        this.tasks.push(task);
+    }
+    stergereTask(task:TODO) {
+    if(this.tasks.length>0) {
+        if(this.tasks.indexOf(task)>-1) {
+            if(task.stare=='nefinalizat'){
+                if(this.tasks.indexOf(task)==0)
+                    this.tasks.splice(this.tasks.indexOf(task),this.tasks.indexOf(task)+1);

```

Comment:
Ai grija la metoda `splice`. In aceasta metoda `array.splice(start[, deleteCount[, item1[, item2[, ...]]]])` , deleteCount este numarul de elemente ce se doresc a fi sterse incepand de la start, deci daca tu folosesti index-ul pentru asta, inseamna ca  vei sterge un numar de elemente egal cu index-ul, incepand de la indexul elementului. 
Ex: 
```
const array = ['A', 'B', 'C', 'D', 'E', 'F'];
array.splice(array.indexOf('C'), array.indexOf('C')); // va sterge incepand de la index-ul lui C, adica 2, un numar de 2 elemente ('C' si 'D')
console.log(array); // va lista ['A', 'B', 'E', 'F'], ceea ce este gresit, pentru ca tu vrei sa stergi un singur element

//corect
array.splice(array.indexOf('C'), 1); // va fi sters doar 'C'
```